src/hb-ot-color-sbix-table.hh

```diff
@@ -116,6 +146,38 @@ struct SBIXStrike
     return hb_blob_create_sub_blob (sbix_blob, glyph_offset, glyph_length);
   }
 
+  bool subset (hb_subset_context_t* c) const
+  {
+    TRACE_SUBSET (this);
+    unsigned int num_output_glyphs = c->plan->num_output_glyphs ();
+
+    auto* out = c->serializer->start_embed<SBIXStrike> ();
+    if (unlikely (!out)) return_trace (false);
+    if (unlikely (!c->serializer->extend (*out, num_output_glyphs + 1))) return_trace (false);
+    out->ppem = ppem;
+    out->resolution = resolution;
+    HBUINT32 head;
+    head = get_size(num_output_glyphs + 1);
+    for (unsigned new_gid = 0; new_gid < num_output_glyphs; new_gid++)
+    {
+      hb_codepoint_t old_gid;
+      if (!c->plan->old_gid_for_new_gid (new_gid, &old_gid) ||
+          unlikely (imageOffsetsZ[old_gid + 1] <= imageOffsetsZ[old_gid] ||
+                    imageOffsetsZ[old_gid + 1] - imageOffsetsZ[old_gid] <= SBIXGlyph::min_size)) {
+        out->imageOffsetsZ[new_gid] = head;
+        continue;
+      }
+      unsigned int delta = imageOffsetsZ[old_gid + 1] - imageOffsetsZ[old_gid];
+      unsigned int glyph_data_length = delta - SBIXGlyph::min_size;
+      if (!(this + imageOffsetsZ[old_gid]).subset(c, glyph_data_length))
+        return_trace (false);
+      out->imageOffsetsZ[new_gid] = head;
+      head += delta;
+    }
+    out->imageOffsetsZ[num_output_glyphs] = head;
+    return_trace (true);
```

Comment:
subset() should return false, if no data should be written for this sub object. So for example if due to the subset of glyphs this particular strike contains no images, then this entire strike can be left out so the subset() method should return false.

Reply:
Done. Should I handle a failure to copy a glyph in that case? My understanding is that the serializer will be in an error state at that point, but should I still try to do cleanup?

Reply:
return false after a failure to copy is reasonable. As you mentioned the serializer will be in an error state so everything will be discarded once you exit the top level subset() call (and typically the operation will be retried with a larger buffer).

Reply:
Ack.